.github/workflows/test-build-rpm.yml

```diff
@@ -191,3 +191,153 @@ jobs:
           from: Opendistro Elasticsearch
           # Optional content type:
           content_type: text/html
+
+
+  Start-ES-Service:
+    needs: [signing-artifacts]
+    name: Start elasticsearch service
+    runs-on: [self-hosted, Linux, X64, amazon]
+    steps:
+      - uses: actions/checkout@v1
+
+      - name: Setup Java
+        uses: actions/setup-java@v1
+        with:
+          java-version: 14
+      
+      - name: Installing python
+        run: sudo yum install python37 -y
+      
+      - name: Getting OD version
+        run: |
+          cd elasticsearch/bin
+          OD_VERSION=`./version-info --od`
+          echo "::set-env name=od_version::$OD_VERSION"
+
+      - name: Starting ES service
+        run: |
+          echo $JAVA_HOME
+          sudo yum install git -y
+          sudo curl https://d3g5vo6xdbdb9a.cloudfront.net/yum/staging-opendistroforelasticsearch-artifacts.repo  -o /etc/yum.repos.d/staging-opendistroforelasticsearch-artifacts.repo
+          sudo yum install wget -y
+          export PATH=$PATH:$JAVA_HOME
+          sudo yum install unzip -y
+          sudo yum install opendistroforelasticsearch-${{env.od_version}} -y
+          sudo chmod 777 /etc/elasticsearch/elasticsearch.yml    
+          sudo /usr/share/elasticsearch/bin/elasticsearch-plugin remove opendistro_security 
+          sudo sed -i /^opendistro_security/d /etc/elasticsearch/elasticsearch.yml
+          sudo sed -i /CN=kirk/d /etc/elasticsearch/elasticsearch.yml
+          sudo sed -i /^node.max_local_storage_nodes/d /etc/elasticsearch/elasticsearch.yml
+          sudo sed -i /^cluster.routing.allocation.disk.threshold_enabled/d /etc/elasticsearch/elasticsearch.yml
+          sudo systemctl start elasticsearch.service
+          sleep 30
+    
+  Test-All-Plugins:
+    needs: [Start-ES-Service]
+    runs-on: [self-hosted, Linux, X64, amazon]
+    steps:
+      - name: Setup Java
+        uses: actions/setup-java@v1
+        with:
+          java-version: 14
+    
+      - name: Installing python
+        run: sudo yum install python37 -y
+
+      - name: Getting OD Version
+        run: |
+          cd elasticsearch/bin
+          OD_VERSION=`./version-info --od`
+          plugin_tag=v$OD_VERSION.0
+          echo "::set-env name=tag::$plugin_tag"
+
+      - name: Checking out ISM repo
+        if: always()
+        uses: actions/checkout@v1
+        with:
+          repository: opendistro-for-elasticsearch/index-management
+          ref: ${{env.tag}}
+
+      - name: IT for ISM
```

Comment:
just keep in mind that sharing cluster might cause some execution issue even if it works well for all of the plugins now

Reply:
Yeah I had that thought! One solution might be clearing up after each test and then installing/starting service again for next plugin
Another solution might be using different self hosted runners for all plugins